samcli/local/apigw/local_apigw_service.py

```diff
@@ -303,6 +303,7 @@ def _request_handler(self, **kwargs):
 
         route = self._get_current_route(request)
         cors_headers = Cors.cors_to_headers(self.api.cors)
+        cors_headers = self._response_cors_headers(request, cors_headers)
```

Comment:
Thanks for creating this PR! Did we need to restrict this behaviour to only run for HTTP APIs? From the docs and the linked issue, this seems to be something that the Lambda function deals with on for REST, but we would need to manage for HTTP.

Reply:
hi @lucashuy 

During the implementation, I referred to [MDN web docs](https://developer.mozilla.org/en-US/docs/web/http/headers/access-control-allow-origin) for reference:

![image](https://user-images.githubusercontent.com/3896985/233853330-2495e26a-10a2-4ff2-81d9-cc165a547bf2.png)

I also referred to the implementation of [middy cors middleware](https://github.com/middyjs/middy/blob/main/packages/http-cors/index.js)

below piece of code will basically parse and return a single origin if matched:
![image](https://user-images.githubusercontent.com/3896985/233853441-21543fb3-6fdc-4920-a8fd-ed763c4074fd.png)

it is explained if looking deeper into the `getOrigin()` function:

![image](https://user-images.githubusercontent.com/3896985/233853485-2eca8eca-36b0-4c53-9c2e-0f5cb921d938.png)

Hope these help explain the reasoning behind my code.

Thanks,